pkg/collector/horizontalpodautoscaler.go

```diff
@@ -135,6 +145,16 @@ func HorizontalPodAutoscaler(cfg config.Config, logger logr.Logger, otelcol v1al
 			metrics = append(metrics, targetCPUUtilization)
 		}
 
+		minReplicas := otelcol.Spec.Autoscaler.MinReplicas
+		if minReplicas == nil {
+			minReplicas = otelcol.Spec.MinReplicas
```

Comment:
I think it is possible that someone could configure `maxReplicas` but not configure `minReplicas`, since this was possible in the earlier versions of the operator's HPA config. So if `minReplicas` is missing then `replicas` should be used as the lower bound instead.  I followed this logic in the validating webhook https://github.com/open-telemetry/opentelemetry-operator/blob/main/apis/v1alpha1/opentelemetrycollector_webhook.go#L93-L102, so should this also be the case here to fall back to `replicas`?

Reply:
Otherwise I think it would still be possible to cause a crash if the original config does not have minReplicas set

Reply:
I see your point. I modified the code to follow an approach similar to the one followed by the defaulter webhook.

Thanks for your feedback!